libs/core/src/test/java/org/elasticsearch/common/CharArraysTests.java

```diff
@@ -70,10 +72,12 @@ public void testConstantTimeEquals() {
         assertTrue(CharArrays.constantTimeEquals(value.toCharArray(), value.toCharArray()));
 
         // we want a different string, so ensure the first character is different, but the same overall length
-        final String other = new String(
-            randomAlphaOfLengthNotBeginningWith(value.substring(0, 1), value.length(), value.length()));
-        assertFalse("value: " + value + ", other: " + other, CharArrays.constantTimeEquals(value, other));
-        assertFalse(CharArrays.constantTimeEquals(value.toCharArray(), other.toCharArray()));
+        final int length = value.length();
```

Comment:
I don't  have strong feeling about it, but might be more straight forward to make sure this never happens on line 70 above.